api/src/main/java/org/openmrs/module/htmlformentryui/tag/UiIncludeTagHandler.java

```diff
@@ -75,12 +131,32 @@ public boolean doStartTag(FormEntrySession formEntrySession, PrintWriter out, No
             }
         }
 
-        return false;
     }
-
-    @Override
-    public void doEndTag(FormEntrySession formEntrySession, PrintWriter out, Node parent, Node node) throws BadFormDesignException {
-        // do nothing
-    }
-
+    
+    /**
+     * Parses urlParameters from a given {@code urlString} into a {@link Map} of properties.
+     * 
+     * @param urlString the {@code URL} to parse
+     * @param formEntrySession HtmlFormSession
+     * @return @link Map} of url parameters
+     * @should handle velocity expressions on FormEntrySession-Contextual {@linkOpenmrsObject}s ie:
+     * * <uiInclude fragment="fragment?patientId=$patient.id" />
+     */
+	protected Map<String, Object> parseFragmentParams(String urlString, FormEntrySession formEntrySession) {
```

Comment:
This method should just become something like:
```java
protected Map<String, Object> paramsToMap(String queryString) {
```
And should be used like this:
```java
paramsToMap(session.evaluateVelocityExpression("..."))
```
Or you should lean towards simplifying the code in that way, as much as possible.